### DISCUSSION TITLE
How to enable traceId in health endpoint?

### DISCUSSION THREAD

**Discussion by filipe-costa** [quarkusio/quarkus#25815] [Q&A] [answered]
At the moment, 

I am using the following libraries:

```
'io.quarkus:quarkus-smallrye-opentracing'
'io.quarkus:quarkus-resteasy-reactive'
'io.quarkus:quarkus-smallrye-health'
...
```

Code example:

```
import org.eclipse.microprofile.health.HealthCheck;
import org.eclipse.microprofile.health.HealthCheckResponse;
import org.eclipse.microprofile.health.Liveness;

import javax.enterprise.context.ApplicationScoped;
import org.jboss.logging.Logger;

@Liveness
@ApplicationScoped
public class CustomHealthCheck implements HealthCheck {

    private final Logger log;

    CustomHealthCheck(Logger log) {
        this.log = log;
    }

    @Override
    public HealthCheckResponse call() {
        log.info("Health check");
        return HealthCheckResponse.up("health check");
    }
}
```

Unfortunately, when logging into the console on this endpoint, I can see that the traceId=%X{traceId} is empty, whereas it works for any other endpoint.

Is there any way to enable this on the health endpoint? If not, is there any workaround?

[a comment by an automated account is not shown]

**Comment by filipe-costa** [comment, marked as answer]
I have noticed that one of the code examples uses the @Traced decorator.

```

import org.eclipse.microprofile.health.HealthCheck;
import org.eclipse.microprofile.health.HealthCheckResponse;
import org.eclipse.microprofile.health.Liveness;
import org.eclipse.microprofile.opentracing.Traced;

import javax.enterprise.context.ApplicationScoped;
import org.jboss.logging.Logger;

@Traced
@Liveness
@ApplicationScoped
public class CustomHealthCheck implements HealthCheck {

    private final Logger log;

    CustomHealthCheck(Logger log) {
        this.log = log;
    }

    @Override
    public HealthCheckResponse call() {
        log.info("Health check");
        return HealthCheckResponse.up("health check");
    }
}

```

Although this works, is there a better way to do this via configuration?

I have noticed that there is a property for small-rye to add to application.properties:

```
quarkus.smallrye-health.context-propagation=true
```

Unfortunately, I don't think this is the correct one because I have used it and it doesn't work.

**Reply by Ladicek** [reply]
Adding `@Traced` is the right thing to do. MicroProfile OpenTracing specifies that all JAX-RS endpoints are traced, so you don't have to add `@Traced` there, but you have to enable tracing explicitly for everything else. Note that a health check is not a JAX-RS endpoint, it's a simple CDI bean.

An important question, however, is: how much value does tracing health checks bring? In my opinion, health checks typically shouldn't reach out to external services, so the value of tracing seems pretty low.

**Reply by filipe-costa** [reply]
Thank you @Ladicek.

At the moment, I don't have enough experience to answer your question. Usually, from what I have seen, that's usually the case. I wonder if there is any valid use case when we have a hard dependency like a Database, and we would like to check if the DB is healthy (ability to perform a connection), or if this should be done and traced in a different system?

Thanks once again !
